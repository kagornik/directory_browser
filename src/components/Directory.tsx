import React from "react";
import { truncateName } from "../utils/customFunction";
import { formContainerStyles, formNameStyles } from "../styles/styles";
import { DirectoryProps } from "../types/types";

const Directory: React.FC<DirectoryProps> = ({
  name,
  icon,
  handleChangeDirectory,
  id,
}) => {
  return (
    <div
      style={formContainerStyles}
      onClick={() => handleChangeDirectory(id) || null}
    >
      <img src={icon} alt="Folder" />
      <span style={formNameStyles}>{truncateName(name)}</span>
    </div>
  );
};

export default Directory;
