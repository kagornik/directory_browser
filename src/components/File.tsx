import React from "react";
import { truncateName } from "../utils/customFunction";
import { formContainerStyles, formNameStyles } from "../styles/styles";
import { FileProps } from "../types/types";

export const File: React.FC<FileProps> = ({ name, icon }) => {
  return (
    <div style={formContainerStyles}>
      <img src={icon} alt="File" />
      <span style={formNameStyles}>{truncateName(name)}</span>
    </div>
  );
};
