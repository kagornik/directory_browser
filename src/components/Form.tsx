import React from "react";
import { truncateName } from "../utils/customFunction";

interface FormProps {
  name: string;
  icon: any;
  handleChangeDirectory: any;
}

const Form: React.FC<FormProps> = ({ name, icon, handleChangeDirectory }) => {
  //   const dispatch = useAppDispatch();

  return (
    <span
      style={{
        marginLeft: "46px",
        marginTop: "25px",
      }}
    >
      <img src={icon} alt="Folder" />
      <span style={{ color: "white" }} onClick={handleChangeDirectory || null}>
        {truncateName(name)}
      </span>
    </span>
  );
};

export default Form;
