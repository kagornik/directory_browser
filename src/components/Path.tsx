import React, { useState, useEffect } from "react";
import { currentDirectoryPathStyles } from "../styles/styles";
import { PathItem } from "./index";
import { useAppDispatch, useAppSelector } from "../utils/hooks";
import { selectDirectoryName, selectDirectoryId } from "../api/selectors";

export const Path: React.FC = () => {
  //to fix
  const [path, setPath] = useState<any>([]);
  const name = useAppSelector(selectDirectoryName);
  const id = useAppSelector(selectDirectoryId);
  useEffect(() => {
    name && setPath([...path, id]);
  }, [name]);

  console.log("path", path);

  return (
    <div style={currentDirectoryPathStyles}>
      {path &&
        path.map((elment: { name: string; id: string }, index: number) => (
          <PathItem
            key={index}
            name={elment.name}
            id={elment.id}
            pathArray={path}
          />
        ))}
    </div>
  );
};
