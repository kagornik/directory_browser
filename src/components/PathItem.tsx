import React from "react";
import { fetchDirectoryFromPath, removePath } from "../api/directorySlice";
import { pathItemStyles } from "../styles/styles";
import { PathItemProps } from "../types/types";
import { useAppDispatch } from "../utils/hooks";

export const PathItem: React.FC<PathItemProps> = ({
  name,
  id,
  pathArray,
  index,
}) => {
  const dispatch = useAppDispatch();
  const firstCondition =
    pathArray[0].name == name || pathArray[pathArray.length - 1].name == name;

  const secondCondition = pathArray[0].name == name;
  const handleChangePath = (id: string | null, index: number) => {
    dispatch(removePath(index));
    dispatch(fetchDirectoryFromPath(id));
  };

  if (pathArray.length > 2) {
    return (
      <span
        style={pathItemStyles}
        onClick={() => {
          handleChangePath(id, index);
        }}
      >
        {!firstCondition ? ` / ${name} / ` : name}
      </span>
    );
  } else {
    return (
      <span
        style={pathItemStyles}
        onClick={() => {
          handleChangePath(id, index);
        }}
      >
        {secondCondition ? name : ` / ${name}`}
      </span>
    );
  }
};
