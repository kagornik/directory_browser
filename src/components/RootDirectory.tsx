import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import folderIcon from "../assets/folder_icon.png";
import fileIcon from "../assets/file_icon.png";
import imageIcon from "../assets/image_icon.png";

const RootDirectory: React.FC = () => {
  const dispatch = useDispatch();

  // useEffect(() => {
  //   dispatch(fetchRootDirectory());
  // }, []);

  return (
    <>
      <div
        style={{
          color: "white",
          fontSize: "20px",
          marginLeft: "46px",
          marginTop: "25px",
        }}
      >
        root
      </div>
      <div>
        <img src={folderIcon} alt="Folder" />
        <img src={fileIcon} alt="Folder" />
        <img src={imageIcon} alt="Folder" />
      </div>
    </>
  );
};

export default RootDirectory;
