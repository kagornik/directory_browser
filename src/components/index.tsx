export * from "./Path";
export * from "./Directory";
export * from "./File";
