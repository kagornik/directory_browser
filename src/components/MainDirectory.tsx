import React, { useEffect } from "react";
import { useAppDispatch, useAppSelector } from "../utils/hooks";
import {
  fetchRootDirectory,
  selectRootFiles,
  selectRootDirectory,
} from "../api/rootDirectorySlice";
import Form from "./Form";
import { currentDirectoryPathStyles } from "../styles/MainDirectoryStyles";
import folderIcon from "../assets/folder_icon.png";
import fileIcon from "../assets/file_icon.png";
import imageIcon from "../assets/image_icon.png";

const RootDirectory: React.FC = () => {
  const dispatch = useAppDispatch();
  const files = useAppSelector(selectRootFiles);
  const directories = useAppSelector(selectRootDirectory);

  const changeDirectory = () => {
    dispatch(fetchRootDirectory());
  };

  useEffect(() => {
    dispatch(fetchRootDirectory());
  }, []);

  return (
    <>
      <div style={currentDirectoryPathStyles}>root</div>
      {directories.map((item, index) => (
        <Form
          key={index}
          name={item.name}
          icon={folderIcon}
          handleChangeDirectory={changeDirectory}
        />
      ))}
      {files.map((item, index) =>
        item.name.includes(".jpg") ? (
          <Form
            key={index}
            name={item.name}
            icon={imageIcon}
            handleChangeDirectory={null}
          />
        ) : (
          <Form
            key={index}
            name={item.name}
            icon={fileIcon}
            handleChangeDirectory={null}
          />
        )
      )}
    </>
  );
};

export default RootDirectory;
