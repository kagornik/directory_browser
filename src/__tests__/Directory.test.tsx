import React from "react";
import { Directory } from "../components/Directory";
import renderer from "react-test-renderer";
import folderIcon from "../assets/folder_icon.png";

it("displays a directory with given name and icon", () => {
  const fn = jest.fn();
  const tree = renderer
    .create(
      <Directory
        name="exampleDirectory"
        id="111"
        handleChangeDirectory={fn}
        icon={folderIcon}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
