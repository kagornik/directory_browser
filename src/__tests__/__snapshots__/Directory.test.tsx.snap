// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a directory with given name and icon 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "display": "inline-block",
      "margin": "40px 0 0 46px",
      "textAlign": "center",
      "width": "50px",
    }
  }
>
  <img
    alt="Folder"
    src="folder_icon.png"
  />
  <span
    style={
      Object {
        "color": "white",
        "display": "block",
        "marginTop": "12px",
      }
    }
  >
    exampleD...
  </span>
</div>
`;
